core/services/run_executor.go

```diff
@@ -121,5 +132,7 @@ func (re *runExecutor) executeTask(run *models.JobRun, taskRun *models.TaskRun)
 
 	input := *models.NewRunInput(run.ID, data, taskRun.Status)
 	result := adapter.Perform(input, re.store)
+	promAdapterCallsVec.WithLabelValues(run.JobSpecID.String(), string(adapter.TaskType()), string(result.Status())).Inc()
```

Comment:
Won't this count get jacked up for the `PendingConfirmations`, which will happen multiple times per run? Maybe we should decrement the previous state?

Reply:
I confirmed w/ @j16r .  These metrics are counted over a timeline so we are tracking changes in the system as opposed to the state of the system.

Reply:
As is this design would count state transitions, so pending confirmations may end up counting more than the number of runs.

If we only wanted to count the total number of jobs that had experienced a PendingConfirmations state, you could subtract whenever a PendingConfirmations -> X state transition had been made.

TBH I'd have both!